keps/sig-cli/kustomize-secret-generator-plugins.md

```diff
@@ -205,52 +266,114 @@ are out of scope for this KEP, however this KEP's
-This may be implemented by the Go team later in which
-case it will just work.
+This is mentioned in the
+[plugin package overview section](https://golang.org/pkg/plugin).
 
-Also, as noted above, someone can write an exec style plugin
+##### mitigation
+
+This KEP offers a framework that dramatically
+lowers the barrier to someone writing an exec style plugin
 which will work on Windows.
 
+Of course, Windows users who want to support
+goplugins could contribute to the golang project.
+
 #### General symbol execution from the plugin
-A Go plugin will be cast to a particular hardcoded interface, e.g.
+
+##### mitigation
+
+A Go plugin, when loaded,
+will be cast to a particular hardcoded interface, e.g.
 
 ```
 type KvSourcePlugin {
-	Get []kvPairs
+	Get []kv.Pair
 }
 ```
 and accessed exclusively through that interface.
 
-#### Existing KV generators continue as an option
-We leave in place the existing three mechanisms
-(_literals_, _env_ and _files_) for generating KV pairs, but
-additionally allow these mechanisms to be expressed as
-`builtin` plugins (see example above).
+### Two means to specify legacy KV generation
+
+The existing three mechanisms and syntax for generating KV
+pairs (_literals_, _env_ and _files_) remain active, but
+additionally the work for this KEP will allow these
+mechanisms to be expressed as `builtin` plugins (see example
+above).
+
+__mitigation:__
 
-If plugins - both Go and other styles - prove unpopular
-or problematic, we can remove them per API change
+If plugins - both goplugins and other styles - prove
+unpopular or problematic, we can remove them per API change
 policies.
 
-If they do prove popular/useful, we can remove deprecate
-the legacy form for (_literals_, _env_ and _files_),
-and help people convert to the new "builtin" form.
+If they do prove popular/useful, we can deprecate the legacy
+form for (_literals_, _env_ and _files_), and help people
+convert to the new `builtin` form to access these KV sources.
 
 ## Graduation Criteria
 
-Many users have been requesting a better way to integrate with secret management tools https://github.com/kubernetes-sigs/kustomize/issues/692
+Many users have been requesting a better way to integrate with
+secret management tools https://github.com/kubernetes-sigs/kustomize/issues/692
+
+* A higher level feature test
+  (like those in [pkg/target](https://github.com/kubernetes-sigs/kustomize/tree/master/pkg/target))
+
+* Documentation of fields in
+  the [canonical example file](https://github.com/kubernetes-sigs/kustomize/blob/master/docs/kustomization.yaml)
 
-* A higher level feature test (like those in [pkg/target](https://github.com/kubernetes-sigs/kustomize/tree/master/pkg/target))
-* Documentation of fields in the [canonical example file](https://github.com/kubernetes-sigs/kustomize/blob/master/docs/kustomization.yaml)
 * Usage [example](https://github.com/kubernetes-sigs/kustomize/tree/master/examples)
 
+* Flag protection on goplugins
```

Comment:
This should be moved out of graduation and into mitigation and risks.

Reply:
done